### PR TITLE
Basic function repair and maintenance.

### PR DESCRIPTION
**Pull request by Relifest**

### CODE REVIEW

File: pytdml/type/__init__.py [hunk 1]
```diff
@@ -29,33 +29,16 @@
...
-from .basic_types import TrainingDataset
-
-
-from .extended_types import EOTrainingDataset
-from .extended_types import EOTrainingData
-from .extended_types import SceneLabel
-from .extended_types import ObjectLabel
-from .extended_types import PixelLabel
-from .extended_types import EOTask
+from .all_types import BaseCamelModel
```
Comment (photonbit) [inline pytdml/type/__init__.py:32]: This is a change that breaks backwards compatibility, we are no longer publishing types that we published before

File: pytdml/type/UiT_HCD_California_2017.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
+{
```
Comment (photonbit) [inline pytdml/type/UiT_HCD_California_2017.json:1]: I might be a good a idea to separate the code and the example files

File: pytdml/type/_utils.py [hunk 3]
```diff
@@ -169,17 +166,34 @@ def _validate_image_format(image_format: str):
         "application/x-netcdf",
         "application/geopackage+sqlite3"
     }
-    return image_format in image_format_list
+    if image_format in image_format_list:
```
Comment (photonbit) [inline pytdml/type/_utils.py:169]: In the branch I was working on, I solved this integrating this logic in the validator, which resulted in a cleaner expression

File: pytdml/type/_utils.py [hunk 4]
```diff
@@ -169,17 +166,34 @@ def _validate_image_format(image_format: str):
...
+        return image_format
+    else:
+        pass
 
 
 def _valid_methods(labeling_methods: str):
     labeling_methods_list = ["manual", "automatic", "semi-automatic", "unknown"]
-    return labeling_methods in labeling_methods_list
+    if labeling_methods in labeling_methods_list:
```
Comment (photonbit) [inline pytdml/type/_utils.py:177]: Same as for the image type, for all the other `_valid` methods in this file

File: pytdml/type/all_types.py [hunk 5]
```diff
@@ -0,0 +1,2307 @@
...
+
+
+class BaseCamelModel(BaseModel):
+    """
+    Basic model with camel case alias
+    Since python use snake case as default
+    We need to convert it to camel case for JSON
+    """
+
+    class Config:
```
Comment (photonbit) [inline pytdml/type/all_types.py:49]: Having a class Config is deprecated in pydantic V2

File: pytdml/type/all_types.py [hunk 6]
```diff
@@ -0,0 +1,2307 @@
...
+        alias_generator = to_camel
+        populate_by_name = True
+        arbitrary_types_allowed = True
+
+    def json(self, **kwargs):
+        data = self.dict(by_alias=True, exclude_unset=True)
```
Comment (photonbit) [inline pytdml/type/all_types.py:55]: Using `dict` is deprecated in pydantic V2, `model_dump` should be used

File: pytdml/type/all_types.py [hunk 7]
```diff
@@ -0,0 +1,2307 @@
...
+        data_without_none = {k: v for k, v in data.items() if v is not None}
```
Comment (photonbit) [inline pytdml/type/all_types.py:56]: I believe this is what the `exclude_none`  option of `model_dump` does

**Review by photonbit — COMMENTED** [review]
I found this merged. I am updating my fork, the work I did to adapt to these changes and open a draft pull request

File: pytdml/type/UiT_HCD_California_2017.json (hunk 2 above)
Comment (Relifest) [reply pytdml/type/UiT_HCD_California_2017.json:1]: Thank you for your reply. I will separate them.

File: pytdml/type/all_types.py (hunk 5 above)
Comment (Relifest) [reply pytdml/type/all_types.py:49]: Thanks for your suggestion, I will change the usage.

File: pytdml/type/all_types.py (hunk 7 above)
Comment (Relifest) [reply pytdml/type/all_types.py:56]: It seems that I still have a limited understanding of using pydantic V2. I will search for more information to modify them. Thanks for your suggestion again.

Comment (photonbit) [reply pytdml/type/all_types.py:56]: I was working towards a full pydantic V2 adoption in the old `basic_types` and `extended_types` files, and left some comments in the pull request: https://github.com/openrsgis/pytdml/pull/17

I think that most of proposals there might be still valid for the `all_types` file

**Pull request merged**